aaelf32/aaelf32.rst

```diff
@@ -0,0 +1,3289 @@
+  +------+------------------+---------+-----------------------+
+  | 108  | R_ARM_TLS_LE32   | Data    | Local Exec Model      |
+  +------+------------------+---------+-----------------------+
+  | 109  | R_ARM_TLS_LDO12  | Arm LDR | Local Dynamic Model   |
+  +------+------------------+---------+-----------------------+
+  | 110  | R_ARM_TLS_LE12   | Arm LDR | Local Exec Model      |
+  +------+------------------+---------+-----------------------+
+  | 111  | R_ARM_TLS_IE12GP | Arm LDR | Initial Exec Model    |
+  +------+------------------+---------+-----------------------+
+
+R_ARM_TLS_GD32 causes two adjacent entries to be added to the dynamically
+relocated section (the Global Offset Table, or GOT).  The first of these is
+dynamically relocated by R_ARM_TLS_DTPMOD32, the second by R_ARM_TLS_DTPOFF32.
+The place resolves to the offset of the first of the GOT entries from the
+place.
+
+R_ARM_TLS_LDM32 is the same as R_ARM_TLS_GD32 except that the second slot in
+the GOT is initialized to zero and has no dynamic relocation.
+
+R_ARM_TLS_LDO32 resolves to the offset of the referenced data object (which
+must be local to the module) from the origin of the TLS block for the current
+module.
+
+R_ARM_TLS_LDO12 is the same as R_ARM_TLS_LDO32 except that the result of the
+relocation is encoded as the 12-bit offset of an Arm LDR instruction.
+
+R_ARM_TLS_LE32 resolves to the offset of the referenced data object (which must
+be in the initial data block) from the thread pointer ($tp).
+
+R_ARM_TLS_LE12 is the same as R_ARM_TLS_LE32 except that the result of the
+relocation is encoded as the 12-bit offset of an Arm LDR instruction.
+
+R_ARM_TLS_IE32 allocates an entry in the GOT that is dynamically relocated by
+R_ARM_TLS_TPOFF32.  The place resolves to the offset of the GOT entry from the
+place.
+
+R_ARM_TLS_IE12GP allocates an entry in the GOT that is dynamically relocated by
+R_ARM_TLS_TPOFF32.  The place resolved to the offset of the GOT entry from the
+origin of the GOT and is encoded in the 12-bit offset of an Arm LDR
+instruction.
+
+.. rubric:: New experimental TLS relocations
+
+TLSDESC_ contains a proposal for enhanced performance of TLS code. At
+this stage the proposal is still experimental, but the relocations
+R_ARM_TLS_DESC, R_ARM_TLS_GOTDESC, R_ARM_TLS_CALL, R_ARM_TLS_DESCSEQ,
+R_ARM_THM_TLS_CALL, R_ARM_THM_TLS_DESCSEQ16 and
+R_ARM_THM_TLS_DESCSEQ32 have been reserved to support this.
+
+.. note::
+   The relocation R_ARM_TLS_DESC re-uses relocation code from the now-obsolete
+   R_ARM_SWI24, but since the former was a static relocation and the new
+   relocation is dynamic there are no practical conflicts in usage.
+
+Dynamic relocations
+~~~~~~~~~~~~~~~~~~~
+
+The dynamic relocations for those execution environments that support only a
+limited number of run-time relocation types are listed in the following table.
+
+.. table:: Dynamic relocations
+  :widths: 10 25 65
```

Comment:
as described above on the version history table, please use a class to specify the width